launcher/config.go

```diff
@@ -291,6 +291,14 @@ func newResource(c *Config) *resource.Resource {
 	}
 	detector := resource.FromEnv{}
 	r, _ := detector.Detect(context.Background())
+
+	hostnameSet := false
+	for iter := r.Iter(); iter.Next(); {
+		if iter.Attribute().Key == conventions.AttributeHostName {
+			hostnameSet = true
```

Comment:
do you need to worry about when it's set but the value is empty?

Reply:
Added a check